tests/testcases.py

```diff
@@ -177,6 +176,17 @@ def run_command(self, args, fake_exit, raise_on_error=True):
             self._clear_stdout()
             return result
 
+    def wait_for_exec_docker(self, timeout):
+        def wait_for_processes():
+            for p in self.exec_docker_processes:
+                if p.returncode is None:
+                    p.communicate()
```

Comment:
Does the `returncode` get set even if you haven't called any follow-up functions like `communicate` on your `Popen` object?